### PR TITLE
Error handling, signalr state and check limit calls before calling API

### PR DESCRIPTION
**Pull request by olalid**
Check if site_state is None before calling its functions.
Add an attribute to online sensors that signals if the signalr stream is connected.
Before calling any API that changes a current limit verify that it is an actual change and skip calling the API if not needed.
Improved error handling in service calls.

### CODE REVIEW

File: custom_components/easee/services.py [hunk 1]
```diff
@@ -261,32 +265,40 @@ async def charger_execute_set_circuit_current(call):
         currentP2 = call.data.get(ATTR_SET_CURRENTP2)
         currentP3 = call.data.get(ATTR_SET_CURRENTP3)
 
-        _LOGGER.debug("execute_service:" + str(call.data))
+        _LOGGER.debug("execute_service: " + str(call.service) + " " + str(call.data))
```
Comment (astrandb) [inline custom_components/easee/services.py:268]: Follow HA code guidelines
```suggestion
        _LOGGER.debug("execute_service: %s %s", str(call.service), str(call.data))
```

File: custom_components/easee/services.py [hunk 2]
```diff
@@ -261,32 +265,40 @@ async def charger_execute_set_circuit_current(call):
...
+            _LOGGER.error("Could not find charger %s", charger_id)
+            raise HomeAssistantError("Could not find charger {}".format(charger_id))
 
     async def charger_execute_set_current(call):
         """Execute a service to set currents for Easee charger."""
         charger_id = call.data.get(CHARGER_ID)
         current = call.data.get(ATTR_SET_CURRENT)
 
-        _LOGGER.debug("execute_service:" + str(call.data))
+        _LOGGER.debug("execute_service: " + str(call.service) + " " + str(call.data))
```
Comment (astrandb) [inline custom_components/easee/services.py:288]: Follow HA code guidelines
```suggestion
        _LOGGER.debug("execute_service: %s %s", str(call.service), str(call.data))
```

File: custom_components/easee/services.py [hunk 3]
```diff
@@ -295,7 +307,7 @@ async def charger_execute_set_charging_cost(call):
         currency = call.data.get(ATTR_COST_CURRENCY)
         vat = call.data.get(ATTR_COST_VAT)
 
-        _LOGGER.debug("execute_service:" + str(call.data))
+        _LOGGER.debug("execute_service: " + str(call.service) + " " + str(call.data))
```
Comment (astrandb) [inline custom_components/easee/services.py:310]: Follow HA code guidelines
```suggestion
        _LOGGER.debug("execute_service: %s %s", str(call.service), str(call.data))
```

**Review by astrandb — CHANGES_REQUESTED** [review]
Looks fine. 
We should make a general review of the attributes at some point. Many of them could be (disabled) diagnostic sensors instead. That would reduce writes to state machine and recorder database. If a user want to monitor these sensors they can easily be enabled. They will also be easier to use in automations. But this is the topic for a separate PR.

File: custom_components/easee/services.py (hunk 1 above)
Comment (olalid) [reply custom_components/easee/services.py:268]: I will probably remove the debug printout entirely...

**Pull request merged**